python/ray/tune/execution/tune_controller.py

```diff
@@ -0,0 +1,973 @@
+import copy
+from collections import defaultdict
+from functools import partial
+from typing import Any, Callable, Dict, List, Optional, Union, Tuple, Set
+
+import logging
+import os
+
+import ray
+from ray.air import Checkpoint, ResourceRequest
+from ray.air.config import CheckpointConfig
+from ray.air._internal.checkpoint_manager import CheckpointStorage, _TrackedCheckpoint
+from ray.air.execution import ResourceManager, PlacementGroupResourceManager
+from ray.air.execution._internal import RayActorManager, TrackedActor
+from ray.tune.error import _AbortTrialExecution
+from ray.tune.execution.ray_trial_executor import _class_cache
+from ray.tune.execution.trial_runner import _TuneControllerBase, TrialRunnerWrapper
+from ray.tune.experiment.trial import (
+    _change_working_directory,
+    _noop_logger_creator,
+    _TrialInfo,
+    _Location,
+    _get_trainable_kwargs,
+)
+from ray.tune.result import TRIAL_INFO, STDOUT_FILE, STDERR_FILE
+from ray.tune.trainable import TrainableUtil
+from ray.tune import TuneError
+from ray.tune.callback import Callback
+from ray.tune.schedulers import TrialScheduler
+from ray.tune.stopper import Stopper
+from ray.tune.search import SearchAlgorithm
+from ray.tune.syncer import SyncConfig
+from ray.tune.experiment import Trial
+from ray.tune.utils import warn_if_slow
+from ray.tune.utils.object_cache import _ObjectCache
+from ray.tune.utils.resource_updater import _ResourceUpdater
+from ray.util.annotations import DeveloperAPI
+from ray.util.debug import log_once
+
+
+logger = logging.getLogger(__name__)
+
+
+@DeveloperAPI
+class TuneController(_TuneControllerBase):
```

Comment:
will we have more tests for it? Right now, only seeing test_actor_caching scenario.